.github/workflows/test_release_packages.yml

```
@@ -0,0 +1,116 @@
+name: Test release packages
+
+on:
+  workflow_dispatch:
+    inputs:
+      today_date:
+        type: string
+
+jobs:
+  generate_release_matrix:
+    runs-on: ubuntu-24.04
+    outputs:
+      release_matrix: ${{ steps.configure.outputs.release_matrix }}
+    steps:
+      - name: Checking out repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Retrieve asset metadata
+        uses: shadowmoose/Github-Release-Data-Action@7cab39a3e1aab55342775ec65908ba28262a3b39
+        id: release_json
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          owner: ROCm
+          repo: TheRock
+          releaseTag: 'nightly-release'
+
+      - name: Generating release matrix
+        id: configure
+        env:
+          ASSET_FILES: ${{ steps.release_json.outputs.data_json }}
+          TODAY_DATE: ${{ inputs.today_date }}
+        run: python ./build_tools/configure_release_matrix.py
+
+
+  test_release_packages:
+    runs-on: ${{ matrix.target.test-runs-on }}
+    needs: generate_release_matrix
+    permissions:
+      contents: write
+    strategy:
+      matrix:
+        target: ${{ fromJSON(needs.generate_release_matrix.outputs.release_matrix) }}
+    env:
+      FILE_NAME: ${{ matrix.target.file_name }}
+      VENV_DIR: ${{ github.workspace }}/.venv
+      THEROCK_BIN_DIR: ${{ github.workspace }}/artifacts/output_dir/bin
+      TAG_NAME: 'nightly-release'
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      # Downloading release tag assets to artifacts directory
+      - name: Download release artifacts
+        uses: robinraju/release-downloader@daf26c55d821e836577a15f77d86ddc078948b05
```

```suggestion
        uses: robinraju/release-downloader@daf26c55d821e836577a15f77d86ddc078948b05 # v1.12
```
Please add the specific version / tag the hash points to as comment.